### DISCUSSION TITLE
Possible to have fbt.param inside fbt.plural?

### DISCUSSION THREAD

**Discussion by alexandernanberg** [facebookincubator/fbt#196] [Q&A] [answered]
Hey! I have a case were I need a param inside a plural, not sure if anything like this is possible? Basically I want to print the name of the episode if singular, otherwise like regular plural. So

`Foo was published` and `3 episodes were published`

I've tried with 👇 but got an `Expected a StringLiteral but got "BinaryExpression" instead` error

```js
fbt(
  fbt.plural(
    fbt.param('episode name', relevantEpisodes[0].name) +
      ' was published',
    relevantEpisodes.length + 1,
    {
      name: 'number of episodes',
      showCount: 'ifMany',
      many: 'episodes were published',
    }
  ),
  'Published episodes summary'
)
```

I can bring out the `fbt.param` outside `fbt.plural` but then the source string are messed up 
```
"{episode name}{number of episodes} episodes were published",
"{episode name}episode was published"
```

**Comment by jrwats** [comment, marked as answer]
Yeah fbt.plural is limited to static strings.  Simplest would be to just use an `if/else`.  Some day... we'll support the parsing of ternaries which would enable you group this into one construct, but I'm not sure it'd be more readable than the below

```js
if (count === 1) {
  return fbt(fbt.param('name', ...) + 'was published', ...);
} else {
  return fbt(fbt.param('count', count) + 'episodes were published', ...);
}
```
